src/main/java/net/minecraftforge/common/Tags.java

```diff
@@ -53,48 +76,19 @@ private static void init(){}
-        public static final TagKey<Block> GLASS_MAGENTA = tag("glass/magenta");
-        public static final TagKey<Block> GLASS_ORANGE = tag("glass/orange");
-        public static final TagKey<Block> GLASS_PINK = tag("glass/pink");
-        public static final TagKey<Block> GLASS_PURPLE = tag("glass/purple");
-        public static final TagKey<Block> GLASS_RED = tag("glass/red");
+        public static final TagKey<Block> GLASS_BLOCKS = tag("glass_blocks");
+        public static final TagKey<Block> GLASS_BLOCKS_COLORLESS = tag("glass/blocks_colorless");
         /**
          * Glass which is made from sand and only minor additional ingredients like dyes
          */
-        public static final TagKey<Block> GLASS_SILICA = tag("glass/silica");
-        public static final TagKey<Block> GLASS_TINTED = tag("glass/tinted");
-        public static final TagKey<Block> GLASS_WHITE = tag("glass/white");
-        public static final TagKey<Block> GLASS_YELLOW = tag("glass/yellow");
+        public static final TagKey<Block> GLASS_BLOCKS_CHEAP = tag("glass/blocks_cheap");
+        public static final TagKey<Block> GLASS_BLOCKS_TINTED = tag("glass/blocks_tinted");
 
         public static final TagKey<Block> GLASS_PANES = tag("glass_panes");
-        public static final TagKey<Block> GLASS_PANES_BLACK = tag("glass_panes/black");
-        public static final TagKey<Block> GLASS_PANES_BLUE = tag("glass_panes/blue");
-        public static final TagKey<Block> GLASS_PANES_BROWN = tag("glass_panes/brown");
-        public static final TagKey<Block> GLASS_PANES_COLORLESS = tag("glass_panes/colorless");
-        public static final TagKey<Block> GLASS_PANES_CYAN = tag("glass_panes/cyan");
-        public static final TagKey<Block> GLASS_PANES_GRAY = tag("glass_panes/gray");
-        public static final TagKey<Block> GLASS_PANES_GREEN = tag("glass_panes/green");
-        public static final TagKey<Block> GLASS_PANES_LIGHT_BLUE = tag("glass_panes/light_blue");
-        public static final TagKey<Block> GLASS_PANES_LIGHT_GRAY = tag("glass_panes/light_gray");
-        public static final TagKey<Block> GLASS_PANES_LIME = tag("glass_panes/lime");
-        public static final TagKey<Block> GLASS_PANES_MAGENTA = tag("glass_panes/magenta");
-        public static final TagKey<Block> GLASS_PANES_ORANGE = tag("glass_panes/orange");
-        public static final TagKey<Block> GLASS_PANES_PINK = tag("glass_panes/pink");
-        public static final TagKey<Block> GLASS_PANES_PURPLE = tag("glass_panes/purple");
-        public static final TagKey<Block> GLASS_PANES_RED = tag("glass_panes/red");
-        public static final TagKey<Block> GLASS_PANES_WHITE = tag("glass_panes/white");
-        public static final TagKey<Block> GLASS_PANES_YELLOW = tag("glass_panes/yellow");
+        public static final TagKey<Block> GLASS_PANES_COLORLESS = tag("glass/panes_colorless");
+
+        public static final TagKey<Block> GLASS_STAINED_BLOCKS = tag("glass/stained_blocks");
```

Comment:
This seems like an unintutive inconsistency. Why are "colorless" listed before "pane"/"block" but "stained" is listed before? Why not do "glass/blocks/stained" and "glass/blocks/colorless"?

Reply:
Good idea with tag intersection. That does solve a pain point I had with these dang glass tags.